src/ray/object_manager/plasma/store.cc

```diff
@@ -286,10 +287,23 @@ uint8_t *PlasmaStore::AllocateMemory(size_t size, MEMFD_TYPE *fd, int64_t *map_s
     *error = PlasmaError::OK;
   }
 
+  // Trigger object spilling if current usage is above the specified threshold.
+  const int64_t footprint_limit = PlasmaAllocator::GetFootprintLimit();
+  if (footprint_limit != 0 && object_spilling_threshold_ < 1.0) {
+    const float allocated_percentage =
+        ((float)PlasmaAllocator::Allocated()) / footprint_limit;
+    if (allocated_percentage > object_spilling_threshold_) {
+      RAY_LOG(INFO) << "Triggering object spilling because current usage "
+                    << allocated_percentage << "% is above threshold "
+                    << object_spilling_threshold_ << "%.";
+      spill_objects_callback_();
```

Comment:
I think instead of invoking spilling directly here, we can just return the status (like `TransientObjectStoreFull`) and make `create_request_queue.cc` handle that?

Reply:
Maybe we can change the status name to something else if that makes sense (`ObjectSpillingRequired`?)

Reply:
I don't think we should do that because we don't want to block object creation on spilling when the object store isn't actually full.

Reply:
Hmm I think it does make sense to keep the callbacks in `create_request_queue` though. How about just returning a bool from the plasma store back to the create queue to spill if we're at the threshold?

Reply:
So we still want to return `Status::ok()` so that the request gets completed, but separately send a flag to the queue to trigger spilling? I'm not sure why we need the extra indirection rather than call the spill callback here.

Reply:
It's just organizational, to keep all OOM-related actions in one place.